pkg/clients/aws/aws.go

```diff
@@ -104,5 +120,67 @@ func ConfigFromFile(file, region string) (*aws.Config, error) {
 		return nil, err
 	}
 
-	return LoadConfig(data, ini.DEFAULT_SECTION, region)
+	return LoadConfig(data, DefaultSection, region)
+}
+
+// String converts the supplied string for use with the AWS Go SDK.
+func String(v string, o ...FieldOption) *string {
```

Comment:
You're right that it is very similar. The main difference is that each implementation wraps the underlying cloud provider's own library, so the advantage of keeping two separate implementations is that if Azure changes their underlying library we'll get that change automatically. There's also a few subtle places where the Azure and AWS packages differ, so I think it could get confusing to (for example) have common string handling code across providers but different string map handling code.

Granted these are doing pretty simple stuff - taking and resolving addresses - so I don't feel strongly about how we approach these helper functions. In general I tend to err toward avoiding "premature deduplication".